nntrainer/src/layers.cpp

```diff
@@ -182,65 +182,55 @@ float ReluPrime(float x) {
-    case Layers::WEIGHT_LECUN_NORMAL: {
-      std::normal_distribution<float> dist(0, sqrt(1 / height));
-      for (unsigned int i = 0; i < width; ++i)
-        for (unsigned int j = 0; j < height; ++j) {
-          float f = dist(gen);
-          W.setValue(0, j, i, f);
-        }
-    } break;
-    case Layers::WEIGHT_LECUN_UNIFORM: {
-      std::uniform_real_distribution<float> dist(-1.0 * sqrt(1.0 / height), sqrt(1.0 / height));
-      for (unsigned int i = 0; i < width; ++i)
-        for (unsigned int j = 0; j < height; ++j) {
-          float f = dist(gen);
-          W.setValue(0, j, i, f);
-        }
-    } break;
-    case Layers::WEIGHT_XAVIER_NORMAL: {
-      std::normal_distribution<float> dist(0, sqrt(2.0 / (width + height)));
-      for (unsigned int i = 0; i < width; ++i)
-        for (unsigned int j = 0; j < height; ++j) {
-          float f = dist(gen);
-          W.setValue(0, j, i, f);
-        }
-    } break;
-    case Layers::WEIGHT_XAVIER_UNIFORM: {
-      std::uniform_real_distribution<float> dist(-1.0 * sqrt(6.0 / (height + width)), sqrt(6.0 / (height + width)));
-      for (unsigned int i = 0; i < width; ++i)
-        for (unsigned int j = 0; j < height; ++j) {
-          float f = dist(gen);
-          W.setValue(0, j, i, f);
-        }
-    } break;
-    case Layers::WEIGHT_HE_NORMAL: {
-      std::normal_distribution<float> dist(0, sqrt(2.0 / (height)));
-      for (unsigned int i = 0; i < width; ++i)
-        for (unsigned int j = 0; j < height; ++j) {
-          float f = dist(gen);
-          W.setValue(0, j, i, f);
-        }
-    } break;
-    case Layers::WEIGHT_HE_UNIFORM: {
-      std::uniform_real_distribution<float> dist(-1.0 * sqrt(6.0 / (height)), sqrt(6.0 / (height)));
-      for (unsigned int i = 0; i < width; ++i)
-        for (unsigned int j = 0; j < height; ++j) {
-          float f = dist(gen);
-          W.setValue(0, j, i, f);
-        }
-    } break;
-    default:
-      W.setZero();
-      break;
+  for (unsigned int i = 0; i < width; ++i) {
+    for (unsigned int j = 0; j < height; ++j) {
+      float f;
+      switch (init_type) {
+        case Layers::WEIGHT_LECUN_NORMAL:
+          f = RandNormal<float>(0, sqrt(1 / height));
+          break;
+        case Layers::WEIGHT_XAVIER_NORMAL:
+          f = RandNormal<float>(0, sqrt(2.0 / (width + height)));
+          break;
+        case Layers::WEIGHT_HE_NORMAL:
+          f = RandNormal<float>(0, sqrt(2.0 / (height)));
+          break;
+        case Layers::WEIGHT_LECUN_UNIFORM:
+          f = RandUniform<float>(-1.0 * sqrt(1.0 / height), sqrt(1.0 / height));
+          break;
+        case Layers::WEIGHT_XAVIER_UNIFORM:
+          f = RandUniform<float>(-1.0 * sqrt(6.0 / (height + width)), sqrt(6.0 / (height + width)));
+          break;
+        case Layers::WEIGHT_HE_UNIFORM:
+          f = RandUniform<float>(-1.0 * sqrt(6.0 / (height)), sqrt(6.0 / (height)));
+          break;
+        default:
+          f = 0;
+          break;
```

Comment:
Isn't this slower? or compilers will do the dirty work?

Reply:
Changed. Thanks.